redhat-developer/web-terminal-tooling
# Remove odo CLI

As of WTO 1.11, Web Terminal is based off of UBI9. odo does not currently officially support RHEL9 based images so it is being removed from the tooling image for the time being.

This is part of https://issues.redhat.com/browse/WTO-298.

Some additional notes:
- The changes to `rh-manifests.txt` were not added in https://github.com/redhat-developer/web-terminal-tooling/pull/71, which is why they are present in this change
- There are 3 other mentions of odo in the codebase, though they do not depend on odo being installed. I opted to keep these as they facilitate re-adding odo in the future if odo ends up supporting RHEL 9. However, maybe it's best to remove these?
  -  https://github.com/redhat-developer/web-terminal-tooling/blob/6a71b0163c1dc0b0f71c5728a7bc31de8d908a07/etc/entrypoint.sh#L26-L27
  - https://github.com/redhat-developer/web-terminal-tooling/blob/6a71b0163c1dc0b0f71c5728a7bc31de8d908a07/etc/initial_config/.zshrc#L50-L52
  - https://github.com/redhat-developer/web-terminal-tooling/blob/6a71b0163c1dc0b0f71c5728a7bc31de8d908a07/etc/get-tooling-versions.sh#L54-L57


# Testing
For convenience, I've pushed the web terminal tooling container built from this PR to `quay.io/aobuchow/wto-tooling:1.11-no-odo`.
1. Launch an OpenShift 4.16+ cluster. I recommend OpenShift 4.17, since the [backport to fix using kubeadmin with WTO](https://github.com/redhat-developer/web-terminal-operator/issues/162#issuecomment-2206400246) has not been merged to OpenShift 4.16 yet. 
2. Install the web-terminal operator from Operator Hub
3. Create a web terminal, click the link to specify a custom image, and provide `quay.io/aobuchow/wto-tooling:1.11-no-odo`. Alternatively, create a web terminal with the default image and run `wtoctl set image quay.io/aobuchow/wto-tooling:1.11-no-odo` in the web terminal console
4. Verify the web terminal starts up successfully.  odo should no longer be found as a command when running `odo` and the output of `help` should not include `odo`

```
Welcome to the OpenShift Web Terminal. Type "help" for a list of installed CLI tools.
bash-5.1 ~ $ odo
bash: odo: command not found
bash-5.1 ~ $ help
Installed tools:
Command    Version         Name
oc         4.16.0          OpenShift CLI
kubectl    1.29.1          Kubernetes CLI
kustomize  5.4.2           Kustomize CLI
helm       3.14.4          Helm CLI
kn         1.12.0          KNative CLI
tkn        0.37.0          Tekton CLI
subctl     0.17.2          Submariner CLI
virtctl    1.2.2           KubeVirt CLI
rhoas      0.53.0          Red Hat OpenShift Application Services CLI
jq         1.6             jq

To customize this terminal, see 'wtoctl'
```
